### PR TITLE
Create LICENSE and fix most awesome-lint errors

### PR DESCRIPTION
**Pull request by zoidbergwill**
Except false positives about sub lists.

Makes some progress on https://github.com/zoidbergwill/awesome-ebpf/issues/22

GitHub doesn't include CC0 in its license templates, so I used the Unlicense which I think is similar.

Compared to before this PR, there are a lot less awesome-lint errors now. The only errors left are:
- `remark-lint:awesome/github` - says GitHub doesn't pick up the License which I think it will when this is merged into master.
- `remark-lint:awesome/list-item` - about awesome-lint not picking up sub-lists, [upstream issue](https://github.com/sindresorhus/awesome-lint/issues/44)

Before:
```
   1 remark-lint:awesome/badge
   1 remark-lint:awesome/github
 136 remark-lint:awesome/list-item
   2 remark-lint:awesome/spell-check
   1 remark-lint:awesome/toc
  51 remark-lint:list-item-bullet-indent
   1 remark-lint:no-dead-urls
```

After:
```
   1 remark-lint:awesome/github
   3 remark-lint:awesome/list-item
  51 remark-lint:list-item-bullet-indent
```

### CODE REVIEW

**Comment by pchaigno** [comment]
GitHub does recognize Creative Commons licenses: https://choosealicense.com/appendix/. If you use the text from choosealicense.com, it should be recognized automatically by [Licensee](https://github.com/licensee/licensee).

File: readme.md [hunk 1]
```diff
@@ -38,45 +36,33 @@ examples to which eBPF brings performance, programmability and flexibility.
...
-    Frequently Asked Questions about contributing to eBPF development.
+- [BPF Documentation](https://www.kernel.org/doc/html/latest/bpf/index.html) - Index for BPF-related documentation coming with the Linux kernel.
+- [linux/Documentation/networking/filter.txt](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/networking/filter.txt) - eBPF specification (somewhat outdated; information should still be valid, but not exhaustive).
+- [BPF Design Q&A](https://www.kernel.org/doc/html/latest/bpf/bpf_design_QA.html) - Frequently Asked Questions on the decisions behind the BPF infrastructure.
+- [HOWTO interact with BPF subsystem](https://www.kernel.org/doc/html/latest/bpf/bpf_devel_QA.html) - Frequently Asked Questions about contributing to eBPF development.
 
 ### Manual Pages
 
--   [`bpf(2)` man page](http://man7.org/linux/man-pages/man2/bpf.2.html) about
+- [`bpf(2)`](http://man7.org/linux/man-pages/man2/bpf.2.html) - Manual pages about
```
Comment (qmonnet) [inline readme.md]: s/pages/page/, there is just one

File: readme.md [hunk 2]
```diff
@@ -38,45 +36,33 @@ examples to which eBPF brings performance, programmability and flexibility.
...
     the `bpf()` system call, used to manage BPF programs and maps from
     userspace.
--   [`tc-bpf(8)` man page](http://man7.org/linux/man-pages/man8/tc-bpf.8.html)
-    about using BPF with tc, including example commands and samples of code.
--   [`bpf-helpers(7)` man page](http://man7.org/linux/man-pages/man7/bpf-helpers.7.html)
-    Description of the in-kernel helper functions forming the BPF standard
-    library.
+- [`tc-bpf(8)`](http://man7.org/linux/man-pages/man8/tc-bpf.8.html) - Manual pages about using BPF with tc, including example commands and samples of code.
```
Comment (qmonnet) [inline readme.md]: s/pages/page/

File: readme.md [hunk 3]
```diff
@@ -85,495 +71,355 @@ examples to which eBPF brings performance, programmability and flexibility.
...
--   [The BSD Packet Filter](https://speakerdeck.com/tuxology/the-bsd-packet-filter)
-    (Suchakra Sharma, June 2017)
-    An introduction mostly covering the tracing aspects.
--   [BPF: tracing and more](http://www.slideshare.net/brendangregg/bpf-tracing-and-more)
-    (Brendan Gregg, January 2017)
-    An introduction mostly covering the tracing aspects.
--   [Linux BPF Superpowers](http://www.slideshare.net/brendangregg/linux-bpf-superpowers)
-    (Brendan Gregg, March 2016)
-    An introduction mostly covering the tracing aspects, first part with flame
+- [A blog post series about eBPF from Ferris Ellis](https://ferrisellis.com/tags/ebpf/)
```
Comment (qmonnet) [inline readme.md]: Missing a description?

File: readme.md [hunk 4]
```diff
@@ -85,495 +71,355 @@ examples to which eBPF brings performance, programmability and flexibility.
...
-    -   [BPF helper functions](https://blogs.oracle.com/linux/notes-on-bpf-2) A review of the kernel functions that can be called from within eBPF programs
-    -   [Communicating with Userspace](https://blogs.oracle.com/linux/notes-on-bpf-3) How BPF communicates with userspace - BPF maps, perf events, bpf_trace_printk
-    -   [Building BPF Programs](https://blogs.oracle.com/linux/notes-on-bpf-4) Setting up your environment to build BPF programs
-    -   [The BPF Bytecode and the BPF Verifier](https://blogs.oracle.com/linux/notes-on-bpf-5) How does BPF ensure that programs are safe?
-    -   [Using BPF to do Packet Transformation](https://blogs.oracle.com/linux/notes-on-bpf-6) One eBPF usage about  packet transformation
+- [IO Visor](https://www.socallinuxexpo.org/sites/default/files/presentations/Room%20211%20-%20IOVisor%20-%20SCaLE%2014x.pdf) - Also introduces [IO Visor project](https://www.iovisor.org/).
+- [BPF — in-kernel virtual machine](https://events.linuxfoundation.org/sites/events/files/slides/bpf_collabsummit_2015feb20.pdf) - Presentation by the author of eBPF.
+- [Extending extended BPF](https://lwn.net/Articles/603983/)
```
Comment (qmonnet) [inline readme.md]: Missing a description?

File: readme.md [hunk 5]
```diff
@@ -85,495 +71,355 @@ examples to which eBPF brings performance, programmability and flexibility.
...
+- [Fast Packet Processing in Linux with AF\_XDP](https://archive.fosdem.org/2018/schedule/event/af_xdp/)
 
 ### bpfilter
 
--   [A high-level post](https://cilium.io/blog/2018/04/17/why-is-the-kernel-community-replacing-iptables/)
-    by Thomas Graf about the motivations behind this project.
--   [A brief introduction](https://qmo.fr/docs/talk_20180316_frnog_bpfilter.pdf)
-    by Quentin Monnet at FRnOG 30.
+- [A high-level post](https://cilium.io/blog/2018/04/17/why-is-the-kernel-community-replacing-iptables/) - About the motivations behind this project.
+- [A brief introduction](https://qmo.fr/docs/talk_20180316_frnog_bpfilter.pdf)
```
Comment (qmonnet) [inline readme.md]: For the two links above we probably want to change the link title. Use the title of the presentations, or at least something more complete like “a brief introduction to bpfilter”? Just “a brief introduction” sounds kind of weird.

File: readme.md [hunk 6]
```diff
@@ -85,495 +71,355 @@ examples to which eBPF brings performance, programmability and flexibility.
...
+    - [Offloading OVS Flow Processing using eBPF](http://openvswitch.org/support/ovscon2016/7/1120-tu.pdf)
+    - [Coupling the Flexibility of OVN with the Efficiency of IOVisor](http://openvswitch.org/support/ovscon2016/7/1245-bertrone.pdf)
+- [Katran](https://code.fb.com/open-source/open-sourcing-katran-a-scalable-network-load-balancer/) - A layer 4 load-balancer based on XDP, open-sourced by Facebook.
+- [XDP in practice: integrating XDP in our DDoS mitigation pipeline](http://netdevconf.org/2.1/session.html?bertin) - Protection against DDoS with XDP at Cloudflare.
+- [Droplet: DDoS countermeasures powered by BPF + XDP](http://netdevconf.org/2.1/session.html?zhou) - Protection against DDoS with XDP at Facebook.
+- [DPDK has a poll-mode driver (PMD) based on AF\_XDP](https://dpdkuserspace2018.sched.com/event/G45Z/dpdk-pmd-for-afxdp)
+- [CETH for XDP](http://www.slideshare.net/IOVisor/ceth-for-xdp-linux-meetup-santa-clara-july-2016) - Common Ethernet Driver Framework for faster network I/O,
     a technology initiated by Mellanox.
--   [The VALE switch](http://info.iet.unipi.it/~luigi/vale/) has
+- [The VALE switch](https://www.unix.com/man-page/freebsd/4/vale/) - has
```
Comment (qmonnet) [inline readme.md]: “has” should be capitalised, and maybe the whole description rephrased?

File: readme.md [hunk 7]
```diff
@@ -85,495 +71,355 @@ examples to which eBPF brings performance, programmability and flexibility.
...
     for executing bpftrace programs in a Kubernetes cluster.
--   [bpfd](https://github.com/genuinetools/bpfd) Framework for running BPF
+- [bpfd](https://github.com/genuinetools/bpfd) - Framework for running BPF
     programs with rules on Linux as a daemon. Container aware.
--   [BPFd](https://github.com/joelagnel/bpfd) A distinct BPF daemon, trying to
+- [BPFd](https://github.com/joelagnel/bpfd) - A distinct BPF daemon, trying to
     leverage the flexibility of the bcc tools to trace and debug remote
     targets, and in particular devices running with Android.
--   [adeb](https://github.com/joelagnel/adeb) provides a Linux shell
+- [adeb](https://github.com/joelagnel/adeb) - Provides a Linux shell
```
Comment (qmonnet) [inline readme.md]: s/Provides//

**Comment by zoidbergwill** [comment]
Thanks for the help with the license, @pchaigno. Replacing it with the text from there and sticking to CC0.

Thanks for your feedback, @qmonnet. Those are some great points, I was worried too with how to better deal with the hyphens between links and descriptions and how to better deal with the long lines.


 > - You might want to choose a style for wrapping lines (pick a max line width and wrap, or never wrap at all) and stick to it for the whole document. Some line breaks look weird now.

So I think wrapping will make diffs cleaner for future edits and additions, but some of our links have long'ish titles which leads to trailing lines with a hyphen, which might be okay. I'll try stick to a sane max line length like 80 or 120 characters.

 > - Some links miss a description, when the title is not super-explicit by itself.

Good point, I'll take a look through them all and try improve the titles or add a good description.

 > - Some link titles or descriptions could maybe be slightly improved. But this is hard to see on the diff to be honest, it's probably better to address this (and previous item) as a follow-up once you have merged these changes.

Great point. It'll definitely be easier to address them in the future. I'll see what I can do in the mean time.

**Comment by zoidbergwill** [comment]
Thanks for your amazing feedback, folks. I've updated the MR, and tried to break it down into saner commits, though the re-formatting and updating the list is still a complicated mess of a commit, unfortunately.

**Comment by qmonnet** [comment]
PR looks good to me. I haven't been through all the details again of the changes on the README, but new style looks consistent and we can probably fix nits (if any) as follow-up.

Licence file looks good to me. Nice to have the yarn.lock update in a separate commit!

**Comment by zoidbergwill** [comment]
Awesome, thanks. I'm gonna go ahead and merge it, and we can fix any nit-picks going forward.

Thanks for all the useful feedback folks.

**Pull request merged**